File: frontend/src/api/notifications.ts
import type { DataResponse, ListResponse } from "../types";
import { http } from "./client";

export interface Notification {
  id: string;
  kind: string;
  title: string;
  message: string | null;
  severity: string;
  entity_type: string | null;
  entity_id: string | null;
  read: boolean;
  created_at: string | null;
}

export const notificationsApi = {
  list: (unread = false) =>
    http.get<ListResponse<Notification>>(`/notifications${unread ? "?unread=true" : ""}`),
  unreadCount: () => http.get<DataResponse<{ unread: number }>>("/notifications/unread-count"),
  markRead: (id: string) => http.post<DataResponse<Notification>>(`/notifications/${id}/read`, {}),
  markAllRead: () => http.post<DataResponse<{ marked: number }>>("/notifications/read-all", {}),
};

File: backend/app/api/v1/notifications.py
"""Founder notification endpoints (Phase 4). Reuses dashboard read scope."""

import uuid

from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.api.deps import get_db, require_permission
from app.domain.roles import Permission
from app.services.notifications import NotificationService

router = APIRouter(prefix="/notifications", tags=["notifications"])
_read = Depends(require_permission(Permission.DASHBOARD_READ))


@router.get("", dependencies=[_read])
def list_notifications(
    unread: bool = Query(default=False), db: Session = Depends(get_db)
) -> dict:
    items = NotificationService(db).list(unread_only=unread)
    return {"data": items, "meta": {"total": len(items)}}


@router.get("/unread-count", dependencies=[_read])
def unread_count(db: Session = Depends(get_db)) -> dict:
    return {"data": {"unread": NotificationService(db).unread_count()}}


@router.post("/{notification_id}/read", dependencies=[_read])
def mark_read(notification_id: uuid.UUID, db: Session = Depends(get_db)) -> dict:
    return {"data": NotificationService(db).mark_read(notification_id)}


@router.post("/read-all", dependencies=[_read])
def mark_all_read(db: Session = Depends(get_db)) -> dict:
    return {"data": {"marked": NotificationService(db).mark_all_read()}}
